src/components/common/PaginatedTxns/index.tsx

```diff
@@ -8,17 +8,32 @@ import useTxQueue from '@/hooks/useTxQueue'
 import PagePlaceholder from '../PagePlaceholder'
 import LoadMoreButton from '../LoadMoreButton'
 import SkeletonTxList from './SkeletonTxList'
+import { TX_LIST_EVENTS } from '@/services/analytics/events/txList'
+import { trackEvent } from '@/services/analytics/analytics'
 
 const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
   const [pageUrl, setPageUrl] = useState<string | undefined>()
   const [allResults, setAllResults] = useState<TransactionListItem[]>([])
   const { page, error, loading } = useTxns(pageUrl)
 
+  const isQueue = useTxns === useTxQueue
+
   useEffect(() => {
     if (page?.results.length) {
-      setAllResults((prev) => prev.concat(page.results))
+      setAllResults((prev) => {
+        const newResults = prev.concat(page.results)
+
+        if (isQueue) {
+          trackEvent({
+            ...TX_LIST_EVENTS.QUEUED_TXS,
+            label: newResults.length,
+          })
+        }
```

Comment:
A side effect in a state setter is no bueno. Suggestion move tracking to its own effect.

```
useEffect(() => {
  trackEvent({
    ...TX_LIST_EVENTS.QUEUED_TXS,
    label: allResults.length,
  })
}, [allResuts.length, isQueue])
```